compiler/resolution/lowerIterators.cpp

```diff
@@ -33,6 +33,31 @@
 #include "symbol.h"
 
 
+// Get the corresponding iterator FnSymbol, insead of:
+//   ...->defaultInitializer->getFormal(1)->type->defaultInitializer
+
+FnSymbol* getTheIteratorFn(Symbol* ic) {
+  return getTheIteratorFn(ic->type);
+}
+
+FnSymbol* getTheIteratorFn(CallExpr* call) {
```

Comment:
This only works if you are calling a function that takes an IC symbol as its first argument.  I would recommend removing function and pushing the argument extraction back to the call sites.
If you keep this function, then you need to add a header comment explaining the conditions under which it can be used.  OR, add an assert that checks that the type of the first argument to the call is an iterator class.